pkg/scanners/scannerv4/vm_convert.go

```diff
@@ -0,0 +1,166 @@
+package scannerv4
+
+import (
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	"github.com/stackrox/rox/generated/storage"
+	pkgCVSS "github.com/stackrox/rox/pkg/cvss"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/protocompat"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/utils"
+)
+
+// ToVirtualMachineScan converts a scan report to the format needed to enrich a virtual machine with scan data.
+func ToVirtualMachineScan(r *v4.VulnerabilityReport) *storage.VirtualMachineScan {
+	return &storage.VirtualMachineScan{
+		ScanTime: protocompat.TimestampNow(),
+		// TODO: find an actual operating system source
+		OperatingSystem: "",
```

Comment:
Propagating comment from #16609 
> Take a look at https://github.com/stackrox/stackrox/blob/master/pkg/scanners/scannerv4/convert.go, which should help fill in some gaps. Related: we already have pkg/scanners/scannerv4 which is a Node and Image Scanner. Can we/are we going to reuse it to be a VMScanner, too? Similarly, placing this code in that path (and potentially reusing functions there)?

Reply:
The target protobuf messages are partial clones of the Image/Node -> Scan -> Component -> Vulnerability ones.
I thought about having wrappers around the messages to have a single conversion logic and proper routing of the values internally, but it might not be the best approach. I'll have a look at #14250